internal/gcsx/syncer.go

```diff
@@ -58,9 +58,14 @@ func NewSyncer(
 		bucket: bucket,
 	}
 
-	appendCreator := newAppendObjectCreator(
-		tmpObjectPrefix,
-		bucket)
+	// Zonal buckets do not currently support Compose, so we always write objects
+	// in their entirety.
+	var composeCreator objectCreator
+	if !bucket.BucketType().Zonal {
+		appendCreator = newAppendObjectCreator(
```

Comment:
This is not correct. https://github.com/GoogleCloudPlatform/gcsfuse-prelaunch/blob/pre-launch-acv2/internal/gcsx/syncer.go#L69